chain/x/oracle/client/common/request_search.go

```diff
@@ -0,0 +1,62 @@
+package common
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	authclient "github.com/cosmos/cosmos-sdk/x/auth/client"
+
+	"github.com/bandprotocol/bandchain/chain/x/oracle/types"
+)
+
+func queryRequest(route string, cliCtx context.CLIContext, rid string) ([]byte, int64, error) {
+	res, height, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s/%s", route, types.QueryRequests, rid), nil)
+	if err != nil {
+		return nil, 0, err
+	}
+	return res, height, nil
+}
+
+func QuerySearchLatestRequest(
+	route string, cliCtx context.CLIContext, oid, calldata, askCount, minCount string,
+) ([]byte, int64, error) {
+
+	events := []string{
+		fmt.Sprintf("%s.%s='%s'", types.EventTypeRequest, types.AttributeKeyOracleScriptID, oid),
+		fmt.Sprintf("%s.%s='%s'", types.EventTypeRequest, types.AttributeKeyCalldata, calldata),
+		fmt.Sprintf("%s.%s='%s'", types.EventTypeRequest, types.AttributeKeyAskCount, askCount),
+		fmt.Sprintf("%s.%s='%s'", types.EventTypeRequest, types.AttributeKeyMinCount, minCount),
+	}
+	searchResult, err := authclient.QueryTxsByEvents(cliCtx, events, 1, 30, "desc")
```

Comment:
Why not use page 0? And if the script has multiple dApp using it (for example, median price oracle), will the limit of 30 be enough -- given that keys can be 100s of symbols requested for DEXes?

Reply:
Ref: https://github.com/cosmos/cosmos-sdk/blob/master/x/auth/client/query.go

Reply:
Because page 0 will lead to error
![Screen Shot 2020-06-09 at 11 09 20](https://user-images.githubusercontent.com/5730246/84105171-b9802280-aa41-11ea-803b-f02e7040bb7c.png)

30 should be enough. It's filtered by oracle script id, calldata, mincount, and askcount. Bad situations can only happen there are 30 exact same requests waiting to be resolved, which is extremely unlikely given current parameter configurations.